### PR TITLE
Implemented `knife user key edit` and `knife client key edit`

### PR DESCRIPTION
**Pull request by tylercloke**
Last PR I swear :) I was trying to keep them smaller for you. This is the last of the knife key commands.

Included in this PR:
+ Small bug fixes to `Chef::Key`
+ Implemented `knife user key edit` and `knife client key edit`
+ Implemented `knife user key show` and `knife client key show`

TODO:
- [x] Get James rolling on the docs. Commands should be pretty solid after we merge this.
+ Docs proposal: https://gist.github.com/tylercloke/896d273232e80de990a7
- [ ] Release and blog post?

Ping @chef/lob @jdmundrawala @danielsdeleo @btm

### CODE REVIEW

**Comment by danielsdeleo** [comment]
@chef/client-core

File: lib/chef/knife/user_key_create.rb [hunk 1]
```diff
@@ -17,7 +17,6 @@
 #
 
 require 'chef/knife'
-require 'chef/knife/key_create_base'
```
Comment (btm) [inline lib/chef/knife/user_key_create.rb]: Mistake to remove this?

**Comment by btm** [comment]
lib/chef/knife/key_update_base.rb and lib/chef/knife/key_edit_base.rb look nearly identical and seem like a good opportunity for a future argument to only be added to one place.

**Comment by tylercloke** [comment]
I started going down that path, but the argument parsing was different enough that it was gonna be annoying. I should probably pull some of the common methods into modules though.

**Comment by tylercloke** [comment]
@btm That's because I renamed the file from `knife_update_base.rb` to `knife_edit_base.rb` to match the rest of the conventions and forgot to uncommit it! `knife_update_base.rb` is rm'ed now :)

**Comment by tylercloke** [comment]
Docs I am planning on adding once this merges and we ship chef-client. Feel free to comment:

https://gist.github.com/tylercloke/896d273232e80de990a7

File: lib/chef/knife/key_edit.rb [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        if @config[:expiration_date]
+          key.expiration_date(@config[:expiration_date])
+        end
+
+        output = edit_data(key)
+        key = update_key_from_hash(output)
+
+        to_display = "Updated key: #{key.name}"
+        to_display = "#{to_display} (formally #{@original_name})" if key.name != @original_name
```
Comment (jdmundrawala) [inline lib/chef/knife/key_edit.rb]: I think you could do `<<` instead. Assigning to the same variable twice in a row looks weird

**Comment by jdmundrawala** [comment]
:+1: You should write stuff in release notes so the world knows about these awesome new commands

**Pull request merged**